.github/actions/call/action.yml

```diff
@@ -91,26 +54,43 @@ runs:
         echo "{version}" | dagger --silent query
         echo "::endgroup::"
 
+    # normal function run
     - name: ${{ inputs.function }}
       shell: bash
+      if: inputs.dev-engine != 'true'
       run: |
-        source /tmp/actions/call/local-envs
-
         set -x
+
         if [[ "${{ inputs.module }}" == "dev" ]]; then
-          # set some sane defaults for the current module
+          flags="--source=.:default"
           if [[ -f $HOME/.docker/config.json ]]; then
-            dagger -m "${{ inputs.module }}" call --source=".:default" --docker-cfg=file:"$HOME/.docker/config.json" ${{ inputs.function }}
-          else
-            dagger -m "${{ inputs.module }}" call --source=".:default" ${{ inputs.function }}
+            flags="$flags --docker-cfg=file:${HOME}/.docker/config.json"
           fi
-        else
-          dagger -m "${{ inputs.module }}" call ${{ inputs.function }}
         fi
+
+        dagger -m ${{ inputs.module }} call ${flags} ${{ inputs.function }}
       env:
         DAGGER_CLOUD_TOKEN: "p.eyJ1IjogIjFiZjEwMmRjLWYyZmQtNDVhNi1iNzM1LTgxNzI1NGFkZDU2ZiIsICJpZCI6ICIwYzhmMGY2Yy00YjY1LTRhODktYTI0YS0yN2NjNWNhNzNmNTcifQ.Em92UDP-KmcNd80Y-euTS2IrRkIE8qTisP3SXYkQi3c"
 
-    - name: Cleanup
+    # nested dev function run
+    - name: ${{ inputs.function }}
       shell: bash
+      if: inputs.dev-engine == 'true'
       run: |
-        rm -rf /tmp/actions/call
+        set -x
+
+        if [[ "${{ inputs.module }}" == "dev" ]]; then
+          flags="--source=.:default"
+          flagsdev="--source=.:default"
+          if [[ -f $HOME/.docker/config.json ]]; then
+            # don't evaluate env vars in internal command...
+            flags="$flags "'--docker-cfg=file:${HOME}/.docker/config.json'
+            # ...but do for the external wrapper
+            flagsdev="$flagsdev --docker-cfg=file:${HOME}/.docker/config.json"
+          fi
+        fi
+
+        cmd="dagger -m ${{ inputs.module }} call ${flags} ${{ inputs.function }}"
+        dagger -m dev call ${flagsdev} dev --target=".:default" --cmd "${cmd}" directory --path=. -o .
```

Comment:
Double-checking this is basically saying "start dev engine as service, run our `dev` module from a container with that service as a dep". I *think* it is but it's inherently a bit mild-melting (not new, using dagger to build+test dagger always has been).

That sounds perfect if so, IIRC the utility of testdev was partially just to ensure our CI still worked on dev engines as opposed to just stable engines, which is what this would result in.

Reply:
Yes :tada: If you have a better way to write this as code, then please shout loudly, it is mind melting as you say :cry: